### PR TITLE
Feature/examples layout

### LINKED ISSUE #22: ngx-components/layout/ has no exported member 'ModalModule'.
## Describe the bug
The Modal package isn't available in the ngx-components node_modules.
Edit to be clear:
The 'layout' bundle consists of a Header, Hero, Modal, ...
But the build package doesn't contain a Modal.

## To Reproduce
1. npm install @acpaas-ui/ngx-components
2. node_modules/@acpaas-ui/ngx-components/layout/index.d.ts doesn't contain Modal

## Expected behavior
The ngx-components layout bundle should contain a Modal module.

## Screenshots
N/A

## OS (please complete the following information):
 - Device: MacBook Pro (Retina, 13-inch, Late 2013)
 - OS: macOS High Sierra (Version 10.13.6)
 - Browser: N/A
 - Version: N/A

## Additional context
ngx-components v1.0.1

### PR DESCRIPTION
**Pull request by TriangleJuice**
## PR Checklist

This PR fulfills the following requirements:

- [x] The commit message follows our guidelines: [Contributing guidelines](https://github.com/digipolisantwerp/acpaas-ui_angular/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] A [changelog entry](https://github.com/digipolisantwerp/acpaas-ui_angular/blob/master/guidelines/CHANGELOG.md) was added

## PR Type

What kind of change does this PR introduce?
* Fixes layout module
* Basic examples for all layout components
* Angular 6.1.x update
* Fixed the modal that wasn't exported

- [x] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
This PR is mostly a fix for the layout module that wasn't  working. Documentation and demos for its submodules will come in separate PRs.

## Resolved issues
Fixes #22

### CODE REVIEW

File: packages/layout/examples/layout/pages/demo/demo.page.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		{
+			href: '/',
+			// icon: 'fa-caret-right',
+			label: 'Label',
+			theme: {
+				slug: 'test',
+				color: 'black',
+				logo: 'https://robohash.org/acpaas-ui',
+			},
+			// items?: SidebarItem[],
```
Comment (TomVanDenBroeck) [inline packages/layout/examples/layout/pages/demo/demo.page.ts]: nog nodig? (zie ook hierboven bij icon)

File: packages/layout/examples/layout/layout.module.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+				href: 'http://cookiepedia.co.uk/all-about-cookies',
+			},
+			cookie: {
+				name: 'cookieconsent_demo',
+				path: '/',
+				domain: '',
+				expiryDays: 1,
+			},
+			elements: {
+				messagelink: `<p id="cookieconsent:desc">{{message}}
```
Comment (lievenvg) [inline packages/layout/examples/layout/layout.module.ts]: No space between curly braces?

File: packages/layout/examples/layout/pages/demo/demo-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	public modalData: any;
+
+	constructor(
+		public modalService: ModalService
+	) {
+		super(modalService);
+	}
+
+	 public submitAndCloseModal() {
+	 	console.log('Submit!');
```
Comment (lievenvg) [inline packages/layout/examples/layout/pages/demo/demo-modal.component.ts]: `console.log`necessary?

File: packages/layout/examples/layout/pages/demo/demo.page.html [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+				<aui-subfooter>
+					<aui-copyright domain="Digipolis"></aui-copyright>
+				</aui-subfooter>
+			</div>
+		</div>
+	</aui-footer>
+</div>
+
+
+<!-- FOOTER -->
```
Comment (lievenvg) [inline packages/layout/examples/layout/pages/demo/demo.page.html]: I think you mean MODAL. 😉

File: packages/layout/examples/layout/pages/demo/demo.page.ts (hunk 1 above)
Comment (TriangleJuice) [reply packages/layout/examples/layout/pages/demo/demo.page.ts]: Yes.

File: package.json [hunk 5]
```diff
@@ -123,7 +123,11 @@
     "tslib": "~1.7.1",
     "tslint": "~5.9.1",
     "typescript": "~2.7.2",
+    "uglifyjs-webpack-plugin": "1.3.0",
     "webpack-bundle-analyzer": "2.13.1",
     "zone.js": "~0.8.26"
+  },
+  "dependencies": {
+    "uglifyjs-webpack-plugin": "1.3.0"
```
[the remark this answers is not in the document]
Comment (TriangleJuice) [reply package.json]: No.

File: packages/layout/examples/layout/pages/demo/demo-modal.component.ts (hunk 3 above)
Comment (TriangleJuice) [reply packages/layout/examples/layout/pages/demo/demo-modal.component.ts]: For now.

File: packages/layout/examples/layout/pages/demo/demo.page.html (hunk 4 above)
Comment (TriangleJuice) [reply packages/layout/examples/layout/pages/demo/demo.page.html]: Indeed.

File: packages/layout/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+## Documentation
+
+Visit our [documentation site](https://acpaas-ui.digipolis.be/) for full how-to docs and guidelines
+
+## Modules
+
+| Name         | Description | Documentation |
+| -----------  | ------ | -------------------------- |
+| `CookieconsentModule` | xxx | [Documentation](./src/lib/cookie-consent/README.md)
```
Comment (jsebrech) [inline packages/layout/README.md]: xxx?

File: packages/layout/examples/layout/pages/demo/demo.page.html [hunk 7]
```diff
@@ -0,0 +1,82 @@
+<h1 class="h3 u-margin-bottom">Layout</h1>
+
+<!-- COOKIE CONSENT -->
+<!-- Just works :-) -->
+
+
```
Comment (jsebrech) [inline packages/layout/examples/layout/pages/demo/demo.page.html]: code snippets?

File: packages/layout/src/lib/cookie-consent/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+# @acpaas-ui/ngx-components/layout
+
+Description goes here.
```
Comment (jsebrech) [inline packages/layout/src/lib/cookie-consent/README.md]: yes, it does go there (zelfde opmerking voor andere plaatsen)

File: packages/layout/src/lib/cookie-consent/README.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+### API
+
+| Name         | Default value | Description |
+| -----------  | ------ | -------------------------- |
+| `@Input() your: string;` | `'API'` | Goes here. |
+
+### Example
+
+```typescript
```
Comment (jsebrech) [inline packages/layout/src/lib/cookie-consent/README.md]: ik snap niet goed hoe je die cookie consent configureert en gebruikt vanuit dit voorbeeld

File: packages/layout/src/lib/footer/README.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+export class AppModule {};
+```
+
+```typescript
+// Controller code
+```
+
+```html
+<!-- HTML code -->
```
Comment (jsebrech) [inline packages/layout/src/lib/footer/README.md]: missing? (zelfde opmerking voor andere plaatsen)

File: packages/layout/src/lib/footer/README.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+## Documentation
+
+Visit our [documentation site](https://acpaas-ui.digipolis.be/) for full how-to docs and guidelines
+
+### API
+
+| Name         | Default value | Description |
+| -----------  | ------ | -------------------------- |
+| `@Input() your: string;` | `'API'` | Goes here. |
```
Comment (jsebrech) [inline packages/layout/src/lib/footer/README.md]: API missing? (zelfde opmerking op andere plaatsen)

**Pull request merged**